.phpstan/phpstan-baseline.neon

```diff
@@ -42,6 +42,11 @@ parameters:
             count: 1
             path: ../src/Admin/AdminHelper.php
 
+        -
+            message: "#^Strict comparison using \\=\\=\\= between null and array\\<T of object\\> will always evaluate to false\\.$#"
+            count: 1
+            path: ../src/Form/DataTransformer/ModelsToArrayTransformer.php
+
```

Comment:
Just saw you added this.

Shouldn't we this instead ?
- Either the phpdoc is wrong
- Either the phpdoc is right and the check is useless. Then we should decide if we remove this in 3.x or in the next major (and so, a NEXT_MAJOR should be added).

Reply:
I won't trust the PHPDoc in this case, because a transformer is mostly used in a *magic* form context which does not use a strict typing system.

The code verifies, if you really pass the correct type.

Reply:
I don't understand. If you don't trust the phpdoc, then change it.

The Phpdoc from Symfony is saying 
```
    /**
     * This method must be able to deal with empty values. Usually this will
     * be NULL, but depending on your implementation other empty values are
     * possible as well (such as empty strings). The reasoning behind this is
     * that data transformers must be chainable. If the transform() method
     * of the first data transformer outputs NULL, the second must be able to
     * process that value.
     *
     * @param mixed $value The value in the original representation
     *
     * @return mixed The value in the transformed representation
     *
     * @throws TransformationFailedException when the transformation fails
     */
    public function transform($value);
```

So handling `null` value seems the right things to do. You should write
```
    /**
     * @param obejct[]|null $collection
     *
     * @return string[]
     *
     * @phpstan-param T[]|null $collection
     */
    public function transform($collection)
```
And you won't get the error.

Reply:
I reverted these things and will create a new PR for the mapper components. This might result in a longer discussion ;)